### ISSUE TITLE
Transfer learning with dlib

### ISSUE THREAD

**Issue by edubois** [davisking/dlib#1163]
Did anybody ever tried to do some transfer learning with dlib?

I'm about to code a ResNet50, and would like to load pre-trained weights.
Thanks for any answer,

**Comment by bakercp** [comment]
Depending on what you mean by [transfer learning](https://cs231n.github.io/transfer-learning/) you can certainly do this in in dlib (e.g. remove the last layer or two from a pre-trained net and train a linear classifier using your own classes and features features generated from the pre-trained dlib net).

But ... it sounds like that isn't what you mean by transfer learning.

It sounds more like you want to build a network using the dlib DNN architecture and import (or deserialize) weights that were trained in another framework (e.g. caffe) and use those weights in your dlib network.

If that's the case, then you might consider the [caffe converter examples](https://github.com/davisking/dlib/tree/master/tools/convert_dlib_nets_to_caffe).  Or it would also be cool if you went all out and just added [NNEF support](https://www.khronos.org/nnef) to dlib 👍

**Comment by edubois** [comment]
Interesting, Ok, that's answering my question. I will see what I can do
with my work.

2018-03-05 21:47 GMT-06:00 Christopher Baker <notifications@github.com>:

**Comment by Entromorgan** [comment]
> It sounds more like you want to build a network using the dlib DNN architecture and import (or deserialize) weights that were trained in another framework (e.g. caffe) and use those weights in your dlib network. @bakercp 
What if I want to rectify an existing dlib DNN architecture like ResNet130 in dnn_face_recognition_ex.cpp，say adding dropout layers for every layers, but I still want to keep the weights imported from dlib_face_recognition_resnet_model_v1.dat. What should I do? 
Thanks in advance.

**Comment by pfeatherstone** [comment]
> Depending on what you mean by [transfer learning](https://cs231n.github.io/transfer-learning/) you can certainly do this in in dlib (e.g. remove the last layer or two from a pre-trained net and train a linear classifier using your own classes and features generated from the pre-trained dlib net).
> 
> But ... it sounds like that isn't what you mean by transfer learning.
> 
> It sounds more like you want to build a network using the dlib DNN architecture and import (or deserialize) weights that were trained in another framework (e.g. caffe) and use those weights in your dlib network.
> 
> If that's the case, then you might consider the [caffe converter examples](https://github.com/davisking/dlib/tree/master/tools/convert_dlib_nets_to_caffe). Or it would also be cool if you went all out and just added [NNEF support](https://www.khronos.org/nnef) to dlib

Adding an ONNX converter tool might be better than NNEF
